### ISSUE TITLE
ETC Core Devs Call 13 & 14

### ISSUE THREAD

**Issue by q9f** [ethereumclassic/ECIPs#362]
# ETC Core Devs Call 14
* When: **Friday, September 25, 2020, 4pm UTC,** 90 minutes max
* Where: [ETC Discord](https://discord.gg/x3WHgrU)
* Focus: Mining Algorithm, Last-Call Proposals, and other 51%-Attack Solutions.

### Agenda

* Last-Call Proposals
    * ~~ECIP 1049 KECCAK256 (Last Call)~~ Moved to call 15 on request
    * [ECIP 1099](https://ecips.ethereumclassic.org/ECIPs/ecip-1099) Calibrate Epoch Duration (Last Call)
* 51% Proposals
  * [ECIP 1092](https://ecips.ethereumclassic.org/ECIPs/ecip-1092) 51-percent attack solution PirlGuard by Callisto (Draft)
  * [ECIP 1094](https://ecips.ethereumclassic.org/ECIPs/ecip-1094) VeriBlock Proof-of-Proof 51%-Attack Prevention (Draft)
  * [ECIP 1096](https://ecips.ethereumclassic.org/ECIPs/ecip-1096) 51% Attack protection system based on Bitcoin Merged Mining (Draft)
  * [ECIP 1097](https://ecips.ethereumclassic.org/ECIPs/ecip-1097) Checkpointing based 51% attack resistance (Draft)
  * [ECIP 1100](https://ecips.ethereumclassic.org/ECIPs/ecip-1100) Modified Exponential Subjective Scoring (Draft)

Live-Agenda: https://hackmd.io/@q9/etc-dev-14

~~_Please post all relevant proposals below that should be discussed and evaluated._~~ Agenda submissions are closed.

# ETC Core Devs Call 13
* When: **Friday, September 11, 2020, 4pm UTC,** 90 minutes max
* Where: Jitsi 
  * Web/App [ETCCoreDevsCall13](https://meet.jit.si/ETCCoreDevsCall13#config.startWithVideoMuted=true&config.startVideoMuted=true)
  * Dial in PIN `1401 7587 98` for [ETCCoreDevsCall13](https://meet.jit.si/static/dialInInfo.html?room=ETCCoreDevsCall13) numbers
* Focus: Mining Algorithm, Last-Call Proposals

### Agenda
* ECIP 1043 Fixed Dag Size Restriction (Last Call)
* ECIP 1099 Calibrate Epoch Duration (Draft, alternative to 1043)
* ECIP 1049 KECCAK256 (Last Call)
* ECIP 1095 SHA3-256 (Last Call, alternative to 1049)

Audio recording: https://www.youtube.com/watch?v=9r_q_z81eh4

**Comment by q9f** [comment]
After recent critique, I will try to host this call on Jitsi as alternative, open-source, and federated conference calling platform.

**Comment by gitr0n1n** [comment]
NO MORE "TBD" agendas @q9f .

Your calls are turning highly controversial because no one knows what you're scheduling, then you push some huge ECIP through like you did at the 8/28 meeting.

**Comment by q9f** [comment]
The goal of this is to crowd-source an agenda as we did in the past. 

I rescheduled the call to have more time to gather proposals and out of respect for the 9/11 victims.

**Comment by gitr0n1n** [comment]
I recommend this PR be closed until the organizer can figure out what he is doing. Scheduling blank meetings on unfavorable dates to many participants is irresponsible. But he should feel free to schedule a meeting that has a SPECIFIC purpose.

**Comment by antsankov** [comment]
Hi, I would like to stick with original plan of a meeting on Sept 11 2020 @ 4 PM UTC. I have invited a number of guests for that time to attend the call, and computed a block activation strategy for this time. This fellow gitr0n1n has said in numerous places that he disagrees with this proposal, and has been harassing everyone associated with the proposal. This is another one of his ploys. Also he did not show up to the previous meeting, he can take a 90 minutes to show up to this one. 

Second of all, I respect GitR0n1n's approach and will happily talk to him directly during the call and address his concerns. However this 9/11 excuse does not justify moving the meeting. I am an american and New Yorker who knows people directly affected, and Friday is a normal working day. So I say we continue.

**Comment by antsankov** [comment]
I will answer all questions that are presented to me on the call, but I am too busy to answer everything posted in the comments. My apologies.

**Comment by developerkevin** [comment]
When someone write TBD. It gives participants a heads up that a meeting is coming and to expect a date/time sometime later. It seems like we are making concessions on behalf of one individual who failed to read the agenda last week and attend the meeting.

These meetings aren't being sneaked around so that certain people don't show up thats just ridiculous. @q9f has done fantastic jobs setting up and leading thees calls. Besides that Ronin is the champion for the RandomX proposal. I've seen no other supporters for RandomX but

**Comment by gitr0n1n** [comment]
@developerkevin please keep your comments on the topic at hand. Thank you, you're welcome to attack me personally in Discord.

**Comment by developerkevin** [comment]
I absolutely do no think we should move it back nearly a week. We should stick with the original date and time, as @antsankov said as well.

**Comment by gitr0n1n** [comment]
perhaps you would rather it be 9/10? or the monday following 9/11?

**Comment by developerkevin** [comment]
No I think we should keep it at Sept. 11, 2020 @ 4 PM UTC.

**Comment by gitr0n1n** [comment]
Has ECIP-1092 scheduled their presentation?

**Comment by gitr0n1n** [comment]
> I will answer all questions that are presented to me on the call, but I am too busy to answer everything posted in the comments. My apologies.

`Last Call` - An ECIP that is done with its initial iteration and ready for review by a wider audience.
`Accepted` - An ECIP that has been in Last Call for at least 2 weeks and any technical changes that were requested have been addressed by the author.

Should we be eating up time on a CDC so you can hold a Q&A on your ECIP-1049 proposal that has so many comments you don't have time to answer them?

Let's think of a productive approach to this:

Can you host your own Q&A outside of the CDC process like other development teams are doing (1097 and 1092)? This is a win/win because it does not drain the limited CDC resources to your proposal and let's ECIP participants focus on the immediate 51% attack solutions that could improve the network right now. And it allows you to host a ECIP-1049 specific Q&A meeting on 9/11.

That feels like a nice accommodation. Format thought: You could read through every comment that is unanswered, Record the response to them on 9/11 for the public to consume and digest. Then your proposal can continue to be reviewed on the October 2nd date that was set on 8/28.

That seems like a fair compromise to the process and the proponents of ECIP-1049. Your proposal would not help the network for at least a year if I am reading correctly. So that is why I am suggesting we prioritize the 51% attack solutions that help in the immediate over long term approaches to network stability like ECIP-1049. @antsankov 

So:
ECIP-1043
ECIP-1092
ECIP-1097

**Comment by gitr0n1n** [comment]
@q9f Can you clarify what the intent of Call 13?

Are we doing Q&A and engaging in `Last Call` for deeper review?
or
Are we voting on `Rejected` or `Accepted` status druing these meeting?

**Comment by gitr0n1n** [comment]
> I will answer all questions that are presented to me on the call, but I am too busy to answer everything posted in the comments. My apologies.

Great @antsankov I am compiling every unanswered question for your team to address during this 9/11 Q&A on ECIP-1049. Right now I count 27 questions, some are multi-pointed questions though. So we should schedule a substantial amount of time for this review as it is not my intent to dominate the entire conversation, but to give your proposal a thorough review for the general public to digest the full ramifications of what you're pushing on the network during these chaotic times.

I've posted two large lists of the questions in the `ecip-1049` and `ecip-1095` channels in discord and pinged you. Please read through those questions prior to the meeting so you're not caught off guard and can adequately address every concern on the call itself.

While I heavily disagree with your proposal and will be there to represent the vocal opposition, I'm looking forward to learning how your team has addressed the adundant criticism of ECIP-1049 in detail. Thank you.

**Comment by gitr0n1n** [comment]
@q9f i'm reading the discussion in #general occurring right now. Since we are hold a 9/11 call. Should we prioritize the 51% solutions to get them moving through the process? I am having a hard time mentally justifying prioritizing ECIP-1049 (a one year plus roll out) over the 51% attack solutions that address the immediate threat to the network:

So high priority ECIPs:
ECIP-1043 https://github.com/ethereumclassic/ECIPs/issues/11
ECIP-1092 https://github.com/ethereumclassic/ECIPs/issues/327
ECIP-1097 https://github.com/ethereumclassic/ECIPs/issues/348

Once these are in motions we have ALL year to discuss moving the netowrk to SHA3, as the network will then be out of the current fire.

Does any of the current ECIP participants feel that the priority should be on these three ECIPs right now and we shouldn't be rushing a miner algorithm change through before addressing these solutions and stabalizing the network?

Maybe my logic is flawed regarding this issue. Feel free to explain why.

**Comment by TheEnthusiasticAs** [comment]
> @q9f i'm reading the discussion in #general occurring right now. Since we are hold a 9/11 call. Should we prioritize the 51% solutions to get them moving through the process? I am having a hard time mentally justifying prioritizing ECIP-1049 (a one year plus roll out) over the 51% attack solutions that address the immediate threat to the network:
> 
> So high priority ECIPs:
> ECIP-1043 #11
> ECIP-1092 #327
> ECIP-1097 #348
> 
> Once these three ECIPs are in motion we have ALL year to discuss moving the network to SHA3, as the network will then be out of the current fire.
> 
> Do any of the current ECIP participants feel that the priority should be on these three 51% attack ECIPs right now and we shouldn't be rushing a miner algorithm change before addressing these solutions and stabilizing the network?
> 
> Maybe my logic is flawed regarding this issue. Feel free to explain why.

The priority is the "51% attack protection", in which the ECIP-1043 #11 is the most favorite at the moment (there are no objections till now).

But sure, there should be no rush. The proposal championings and discussions can run parallel, if the capacity of the developers/network participants allow it.

**Comment by q9f** [comment]
**ECIP-1043:** Just some findings regarding the DAG size.
* ethminer 0.17.0 on my 4GB Radeon RX 470 is limited to epoch 351 (3.74GB)
* claymore 15.0.0 on my 4GB Radeon RX 470 is limited to epoch 351 (3.74GB)
* teamredminer 0.7.10 on my 4GB Radeon RX 470 is limited to epoch 382 (3.98GB)

The theoretical possible limit should be epoch 383 (3.99GB), however, we didn't manage to find a miner that is capable to actually deal with that.

Current epoch is 372 (3.91GB) at the time of writing.

So we have a soft limit of around 351 and a hard limit of 383. Therefore, we already entered the critical phase of DAG size.

> B: (current spec) stop growth at block 10,000,000 (DAG is about 3.6GB then)

This really needs to be updated. It looks like this is no longer a viable option.

**Comment by Dexaran** [comment]
@q9f I would suggest to test [Phoenix Miner 5.1c](https://phoenixminer.org/) as it seem to be the most effective from my previous experience.

**Comment by q9f** [comment]
Phoenix does not work on my operating system.

**Comment by q9f** [comment]
**ECIP-1049:** Just some findings on interoberability - I did some research on the implications of vanilla Sha3 and I would strongly encourage to go for Keccak256 rather than Sha3. 

The main reason is having an OPCODE `0x21 SHA3` is very unlikely to convince the Ethereum Foundation community to realize and accept. Going solo with such an OPCODE on Ethereum Classic would mean we would have to fork all tooling including Solidity compiler. 

As an alternative, we could implement a SHA3 precompile but that's also unlikely on Ethereum Foundation network. That way we might _lose_ the freshly gained _protocol parity_ indefinitely. Adopting Keccak256 does not only reduce the complexity but also retains highest compatibility between these ecosystems without much effort as the OPCODE `0x20 KECCAK` already exists for both chains and no further tooling would be required.

**Comment by shanejonas** [comment]
just a generic concern around adding opcodes, if we added `0x21 SHA3` and ETH adds `0x21` as something else then we will face some issues with tooling across the ecosystem as well.

**Comment by q9f** [comment]
> just a generic concern around adding opcodes, if we added 0x21 SHA3 and ETH adds 0x21 as something else then we will face some issues with tooling across the ecosystem as well.

That's exactly what I tried to say. ref https://github.com/ethereum/EIPs/issues/2951

**Comment by zmitton** [comment]
meeting was just moved to Discord because of a security flaw in jitsi

**Comment by q9f** [comment]
^

**Comment by shanejonas** [comment]
sucks jitsi didn’t work out but the moderation in the other discord prevents many of the team from joining the call.

**Comment by q9f** [comment]
We can make Jitsi work if we have our own server. This enables us proper moderation roles and tools as far as I understand.

I'm sorry that I didn't look into that earlier.

**Comment by q9f** [comment]
TL;DR of decisions:

* ECIP 1043 will be **replaced** by ECIP 1099
* ECIP 1099 will be moved to **last call** (3 weeks)
* ECIP 1049 will remain in **last call** (another 4 weeks)
* ECIP 1095 will be **rejected** in favor of keccak256 proposal ECIP 1049

**Comment by developerkevin** [comment]
nobody was prevented from joining today's call. quite the contrary. we had many trolls find their way in and spam  the call with annoying sounds. not sure what was preventing you from joining as etclabs core devs are not banned. 🤔

**Comment by slavserver** [comment]
ECIP 1092 , в первую очередь.. а потом уже можно думать об усовершенствовании... но нет же, уже больше месяца тянетя кота за яйца

**Comment by gitr0n1n** [comment]
> @q9f i'm reading the discussion in #general occurring right now. Since we are hold a 9/11 call. Should we prioritize the 51% solutions to get them moving through the process? I am having a hard time mentally justifying prioritizing ECIP-1049 (a one year plus roll out) over the 51% attack solutions that address the immediate threat to the network:
> 
> So high priority ECIPs:
> ECIP-1043 #11
> ECIP-1092 #327
> ECIP-1097 #348
> 
> Once these three ECIPs are in motion we have ALL year to discuss moving the network to SHA3, as the network will then be out of the current fire.
> 
> Do any of the current ECIP participants feel that the priority should be on these three 51% attack ECIPs right now and we shouldn't be rushing a miner algorithm change before addressing these solutions and stabilizing the network?
> 
> Maybe my logic is flawed regarding this issue. Feel free to explain why.

After listening to the end of the [CDC 13](https://youtu.be/9r_q_z81eh4?t=3422). It is disheartening to hear that developers haven't had the time to even review ECIP-1092 at this point due to the `mining algo change` narrative being pushed ahead of legitmate 51% attack solutions. I ask the ECIP participants to stay focused on the `high priority` solutions like my previous messages have been suggesting. We don't have the bandwidth to be fighting about a mining algo change right now. 51% attack solutions should be on the forefront of our efforts.

In the spirit of prioritizing `51% Attack Solutions` over all else, I have requested to move ECIP-1093 to `Withdrawn` status. https://github.com/ethereumclassic/ECIPs/pull/376

**Comment by st0ckiz** [comment]
Hello, I just have to lift the opposition to the Algo change that some people wants to see here.

In my eyes, this is very clear; and i don't know who you panic and cry for a CHANGE of algo.

We have to keep in mind that a BIG part of ETH mining is ASIC rigs, we know that there are some different ASIC creators that can resume the production if its needed.

WE ALSO KNOW that ETH have a thing called ProgPOW in the making, wouldnt this alone make 51% attacks nearly impossible???


What im saying is straight forward and not hard to get.

Dont change algo, you will regret it. Upgrade the system to prevent 51%:s, but dont miss out on the big network change with ETH 2.0 Rollup!!!

**Comment by TheEnthusiasticAs** [comment]
I am suggesting to make the Call 14, 25th of September, on ETC discord as well: https://discord.gg/3J7ztm. @q9f I need your confirmation to update your first post. Thank you.

**Comment by q9f** [comment]
Yes, let's do discord.

**Comment by gitr0n1n** [comment]
> I am suggesting to make the Call 14, 25th of September, on ETC discord as well: https://discord.gg/3J7ztm. @q9f I need your confirmation to update your first post here. Thank you.

Your link is dead. I believe this is an active invite: https://discord.gg/hQs894U

**Comment by gitr0n1n** [comment]
Given the ECIP-1049 has dominated CDC 11, 12, and 13. It's probably great to allow other proposals to be addressed with the limited 90 minute call on CDC 14.

I've scheduled a 120 min max CDC 15 specifically to talk about ECIP-1049 only and the author has agreed to show up to that call. So continued conversation on ECIP-1049 can likely wait a week to let all these other proposals have a chance to be discussed.
https://github.com/ethereumclassic/ECIPs/issues/382

**Comment by TheEnthusiasticAs** [comment]
> > I am suggesting to make the Call 14, 25th of September, on ETC discord as well: https://discord.gg/3J7ztm. @q9f I need your confirmation to update your first post here. Thank you.
> 
> Your link is dead. I believe this is an active invite: https://discord.gg/hQs894U. I've updated a PR for the website's blog.

Yes, @gitr0n1n, it got invalid. I took your shared one. cc @q9f

**Comment by antsankov** [comment]
I won't be able to speak at this discussion, but will be able to listen. However on October 2nd @ 4PM UTC, I will be doing a full call with GitR0nin that will be devoted entirely to the subject of ECIP-1049 and Keccak256.  

Ideally this call can be entirely focused on PirlGuard and Mess, and which one is a better approach.

**Comment by TheEnthusiasticAs** [comment]
The 14th's call link was updated again.

**Comment by bobsummerwill** [comment]
Please can we add "ECIP 1097: Checkpointing based 51% attack resistance (Draft)" to the agenda, @q9f?
Thanks!

**Comment by gitr0n1n** [comment]
Good point @bobsummerwill I also see 1094 was proposed, but not sure where that one is at.

## Agenda

51% Proposals
* [ECIP 1092](https://ecips.ethereumclassic.org/ECIPs/ecip-1092) 51-percent attack solution PirlGuard by Callisto (Draft)
* [ECIP 1094](https://ecips.ethereumclassic.org/ECIPs/ecip-1094) VeriBlock Proof-of-Proof 51%-Attack Prevention (Draft)
* [ECIP 1096](https://ecips.ethereumclassic.org/ECIPs/ecip-1096) 51% Attack protection system based on Bitcoin Merged Mining (Draft)
* [ECIP 1097](https://ecips.ethereumclassic.org/ECIPs/ecip-1097) Checkpointing based 51% attack resistance (Draft)
* [ECIP 1099](https://ecips.ethereumclassic.org/ECIPs/ecip-1099) Calibrate Epoch Duration (Last Call)
* [ECBP 1100](https://ecips.ethereumclassic.org/ECIPs/ecip-1100) Modified Exponential Subjective Scoring (Draft)

**Comment by q9f** [comment]
ok, removed 1049 and added the other proposals :+1:

**Comment by SergioDemianLerner** [comment]
Thanks for adding ECIP 1096. We'll be there to present and respond to questions. What is your idea? How much time will be assigned to ECIP 1096 ?

**Comment by q9f** [comment]
Updated the block numbers for 1099:

* For the Ethereum Classic main network: `ETCHASH_FORK_BLOCK := 11_700_000` (Epoch 390)
* For the Mordor Classic test network: `ETCHASH_FORK_BLOCK := 2_520_000` (Epoch 82)

Which basically means 4 weeks for Mordor and 8 weeks for the Classic main network.

We will reevaluate this in tomorrow's call.

**Comment by q9f** [comment]
> Thanks for adding ECIP 1096. We'll be there to present and respond to questions. What is your idea? How much time will be assigned to ECIP 1096 ?

So, first thing will be finalizing ECIP 1099 which is in last call and we need to update the block numbers to allow for Besu to implement the changes and second part will be discussing the "51% proposals."

In the end we will have to decide for one as we _most likely_ will not implement all of them and the core devs have to evaluate if it's sufficient to just implement one protective measure or whether we need multiple.

The community sentiment appears to either favor 1092 "Pirlguard" by Callisto or the 1100 "MESS" proposal by the ETC Core working group. The discussion on 1092 was cut off last time, unfortunately, so we will start with that and also allow for comparing with the other proposals.

What we could do, because it's five different proposals, that I ask 3-4 questions that each proposal champion has to answer and this will allow us to create some matrix of what the proposals do, what they don't do, and what they require. I'll come up with something by tomorrow.

**Comment by realcodywburns** [comment]
Will the Berlin HF eips be included in this fork or will there be another later fork for bls12-381 and simple subroutines? 
Or do we need a thread for the HF separate from this stream?

**Comment by bobsummerwill** [comment]
@realcodywburns There has been no discussion about ETH Berlin changes for ETC to my knowledge.
I doubt that there are any kind of priority for them given everything else we have on our plate.

**Comment by developerkevin** [comment]
I'm unsure if we should have the Veriblock ECIP on the table for this call as they have not presented their proposal, but plan to. I've been in contact with Max.

He says, _"We are writing up a ETC-focused "what is PoP and how will it protect ETC" layman's explanation article which we want to publish prior to the call [presentation]."_

Based on the fact they haven't presented, although having much time to do so, I'm unsure if it would be fair to them to measure a consensus on moving their ECIP either way in the process. They plan on presenting their solution mid-next week as noted by Max here: https://github.com/ethereumclassic/ECIPs/issues/360#issuecomment-698594259

To be honest based on what I've seen I see the MESS solution as being the most effective out of the bunch. It also seems it would be much easier to implement into Besu given Chainsafe's affiliation in its creation. Mess seems much less complex and more in line with ETC's decentralized manifesto and declaration of independence, more than nearly all the others which either introduce layers of unnecessary complexity i.e. through new mining mechanisms, adding validators which will require some sort of on-chain governance mechanism (also we've maintaining a network with validators become a bit difficult on Kotti), or relying on a trusted third party whether it be another  network or validators.

By the way thanks for coordinating this call Afri! .

**Comment by realcodywburns** [comment]
@bobsummerwill in your absence prior to the attacks we had initial discussions on berlin. Then it was delayed on eth due to geth being the majority client and was postponed until the fall for eth. But it looks to still be un planned for eth so no real rush.

**Comment by q9f** [comment]
> What we could do, because it's five different proposals, that I ask 3-4 questions that each proposal champion has to answer and this will allow us to create some matrix of what the proposals do, what they don't do, and what they require. I'll come up with something by tomorrow.

I like to see the following questions answered during the call (1 min/proposal max):
* does it require a hardfork? (backwards-compatibility)
* is there a specification for ethereum classic available? (some proposals come from other ecosystems)
* is there a transition for activation specified? (does it need one? activation block? smooth transtion?)
* is there any implementation available? (python, go, crystal, c++, rust, ...)
* are test cases available that can be used by the client teams?

**Comment by SergioDemianLerner** [comment]
ECIP 1096 :
1. does it require a hardfork? (backwards-compatibility)
Yes, to get 100% of the protections the system provides.

2. does it break nakamoto consensus? (one cpu, one vote)
No.

3. is there a specification for ethereum classic available? (some proposals come from other ecosystems)
The ECIP provides guidance on implementation, but it's not 100% complete. We could finish it in a short time if the ETC community is really interested in moving forward.

4. is there a transition for activation specified? (does it need one? activation block? smooth transtion?)
We haven't yet specified. It can be smooth or drastic. The proposal enables the level protection to be gradually incremented.

5. is there any implementation or simulation available? (python, go, crystal, c++, rust, ...)
are test cases available that can be used by the client teams?
No.

I would also like to mention that ECIP 1096 provides some key benefits, such being a decentralized solution.
And it could also allow to build pre-compiles in ETC to access the Bitcoin headers for further interoperability, as the Bitcoin headers will be attached to ETC headers in an censor-resistant way.

**Comment by SergioDemianLerner** [comment]
To me more precise on (1).

If ETC wants the solution to be fully decentralized, then some part of the subsidy must be channeled to visibility proof miners by a hard fork. 
The payments can also come from a treasury (i. .e ecip-1098). In that case the best implementation is having a federation (which can be eithers RSK current federation or IOHK proposed federation) to issue the visibility proofs.

**Comment by q9f** [comment]
Thanks everyone. Kevin has a recording.

Here are my rough notes:

* ECIP 1049 was skipped, **remains in "last call"** for another two weeks, please join the breakout call next week to further discuss keccak mining algorithm, see #382 
* **ECIP 1099 remains in "last call"** but new block numbers (pushed back to allow besu to implement it), this will move to "accepted" next friday unless you raise your concerns, core developers start working on implementations and preparing releases, see #371
* **ECBP 1100** was demonstrated on the "messnet" testnet, it **moves to "active"** with a mordor block number for testing, mainnet decision depends on testnet progress, no mainnet block number yet
* ECIP 1092 was skipped, nobody spoke in favor of it
* ECIP 1094 was skipped, there will be a presentation by max next week
* ECIP 1096 was discussed
* ECIP 1097 was discussed, requires either a hardfork or optinally a soft fork, we would have to chose trusted parties for committee

Important items are **bold**.

**Comment by developerkevin** [comment]
@SergioDemianLerner I have a question regarding your comment below.

> The payments can also come from a treasury (i. .e ecip-1098). In that case the best implementation is having a federation (which can be either RSK's current federation or IOHK proposed federation) to issue the visibility proofs.

Hypothetically if there was some collaboration between the two, would there still be a required tax on mining rewards to pay for "visibility proof miners?" I recall you saying it is possible to use your solution without the need to give any portion of the miner's block rewards to visibility proof miners—would that make this proposal non viable? Would the subsidy provided be temporary and what are the reciprocations of not providing any subsidy to them?

To be honest, I don't think taking away a portion of block rewards to subsidize third party miners in order to temporarily protect the network will sit well with the people in ETC, especially the ETC miners and given the many other solutions that don't involve such a mechanism and incentive to outside miners.

**Comment by TheEnthusiasticAs** [comment]
An audio recording for the Call 14 was added in the first post (q9f).
